impl_core/src/main/java/io/opencensus/implcore/trace/propagation/TraceContextFormat.java

```diff
@@ -123,8 +123,7 @@
       // TODO(bdrutu): Do we need to verify that version is hex and that for the version
       // the length is the expected one?
       checkArgument(
-          traceparent.charAt(TRACE_OPTION_OFFSET - 1) == TRACEPARENT_DELIMITER
```

Comment:
Could you add a unit test reproducing the issue #2038 and confirming it's fixed by this change?

Reply:
I haven't find a recurrence here, but anyway, there is something wrong with the logic of this code, right?

Reply:
Unfortunately, I'm not familiar with this code.

I do agree that the expression seems to have a redundant duplicate condition, though removing the first check may lead to some other issue since when it's "false" the remaining conditions won't get executed.
So I suggest to add a test reproducing the issue this change is supposed to fix.